src/Projects/Modules/HorribleSubs/Server/Module.HorribleSubs.Cida/Services/SearchService.cs

```diff
@@ -0,0 +1,106 @@
+        {
+            var webClient = new WebClient { Proxy = null };
+            var page = await webClient.DownloadStringTaskAsync(new Uri($"{BaseUrl + searchTerm}")).ConfigureAwait(false);
+            return this.ParseResultPage(page).ToArray();
+        }
+
+        public SearchResult[] Search(string searchTerm)
+        {
+            var webClient = new WebClient { Proxy = null };
+            var page = webClient.DownloadString(new Uri($"{BaseUrl + searchTerm}"));
+            return this.ParseResultPage(page).ToArray();
+        }
+
+        private IEnumerable<SearchResult> ParseResultPage(string page)
+        {
+            var lines = page.Split(new[] { '\n' }, StringSplitOptions.RemoveEmptyEntries);
+            return lines.Select(
+                x =>
+                {
+                    var line = x.Substring(x.IndexOf('{'), x.IndexOf('}') - x.IndexOf('{') + 1);
+                    var data = this.ParseJavaScriptObject(line);
+                    return new SearchResult()
+                    {
+                        BotName = data.Bot,
+                        FileSize = data.Size,
+                        PackageNumber = data.Number,
+                        FileName = data.File
+                    };
+                });
+        }
+
+        private JsonData ParseJavaScriptObject(string line)
+        {
+            const string bIdentifier = "b:\"";
+            const string nIdentifier = "n:";
+            const string sIdentifier = "s:";
+            const string fIdentifier = "f:\"";
+            var regex = new Regex("([b|n|s|f]:\\\"?.*?\\\"?[,|\\}])");
+            var result = new JsonData();
+            foreach (var match in regex.Matches(line))
+            {
+                var stringMatch = match.ToString();
+                stringMatch = stringMatch[0..^1];
+                if (stringMatch.Contains(bIdentifier))
+                {
+                    stringMatch = stringMatch.Replace(bIdentifier, string.Empty);
+                    stringMatch = stringMatch[0..^1];
+                    result.Bot = stringMatch;
+                }
+                else if (stringMatch.Contains(nIdentifier))
+                {
+                    stringMatch = stringMatch.Replace(nIdentifier, string.Empty);
+                    result.Number = long.Parse(stringMatch);
+                }
+                else if (stringMatch.Contains(sIdentifier))
+                {
+                    stringMatch = stringMatch.Replace(sIdentifier, string.Empty);
+                    result.Size = long.Parse(stringMatch);
+                }
+                else if (stringMatch.Contains(fIdentifier))
+                {
+                    stringMatch = stringMatch.Replace(fIdentifier, string.Empty);
+                    stringMatch = stringMatch[0..^1];
+                    result.File = stringMatch;
+                }
+            }
+
+            return result;
+        }
```

Comment:
Maybe write a custom json-converter for this one? At least it seems like this is trying to parse the json into an object?
That would be a bit more "understandable".

Reply:
I don't want to use anything with JSON bc HorribleSubs is not sending back JSON valid data. They are sending valid JS code. For context:
They are creating an array `p.k[]` and then starting the call to that endpoint. The endpoint is giving back smth like this:
```p.k[0] = {b:"Ginpachi-Sensei", n:421, s:1843, f:"[Mew] Sword Art Online - Alicization - 01 (BDRip Dual Audio 1080p) [189BEE6A].mkv"};
p.k[1] = {b:"Ginpachi-Sensei", n:422, s:832, f:"[Mew] Sword Art Online - Alicization - 02v2 (BDRip Dual Audio 1080p) [121CA949].mkv"};
p.k[2] = {b:"Ginpachi-Sensei", n:423, s:842, f:"[Mew] Sword Art Online - Alicization - 03 (BDRip Dual Audio 1080p) [08C18851].mkv"};```
What they do is, take the response and just put it into an `eval(responseText)` which executes the JS there. What i'm doing there is parsing the JS not JSON. Even the object creation is not valid JSON according to the RFC therefor i don't want to connect JSON with that in any way

Reply:
Oh no